test/test_object_id.cpp

```diff
@@ -211,6 +212,34 @@ TEST(ObjectId_PrimaryKey)
     }
 }
 
+TEST(ObjectId_Commit)
+{
+    // Tend to discover errors in the size calculation logic
```

Comment:
May be worth running a a loop to insert enough data that it causes some splits in the table.

Reply:
Well - it was this exact test case that revealed the error.